### PR TITLE
Custom contacts e2e

### PR DESCRIPTION
**Pull request by newtewt**
# Description

Tests for custom contact persons and places.

https://github.com/medic/cht-release-testing/issues/61

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/contacts/new-lineage-structure.wdio-spec.js [hunk 1]
```diff
@@ -7,12 +6,12 @@ const commonPage = require('../../page-objects/common/common.wdio.page');
 const utils = require('../../utils');
 const sentinelUtils = require('../sentinel/utils');
 
-const centerName = faker.address.city();
-const centerContact = faker.name.findName();
-const area = faker.address.city();
-const areaContact = faker.name.findName();
-const household = faker.address.city();
-const householdContact = faker.name.findName();
+const centerName = 'Franklin';
```
Comment (meghna-khemka) [inline tests/e2e/contacts/new-lineage-structure.wdio-spec.js:9]: We have a GitHub issue tracking the search fix, can we reference it here as a comment to indicate we need to come back and use faker again once that issue is resolved

Comment (newtewt) [reply tests/e2e/contacts/new-lineage-structure.wdio-spec.js:9]: Based on our discussion during standup. We are going to remove faker and work on implementing a secondary suite to run with mocked data.

File: tests/e2e/contacts/add-custom-hierarchy.wdio-spec.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    await commonPage.goToPeople();
+    await expect(await contactsPage.leftAddPlace()).toHaveText(translations[customTopLevel.create_key]);
+  });
+
+  it('the LHS add place button should read add place when multiple places exist at the current level', async () => {
+    const settings = await utils.getSettings();
+    settings.contact_types.push(secondTopLevel);
+    const forms = customTypeFactory.formsForTypes([secondTopLevel]);
+    await utils.revertSettings(true);
+    await utils.updateSettings({ contact_types: settings.contact_types }, true);
```
[the remark this answers is not in the document]
Comment (newtewt) [reply tests/e2e/contacts/add-custom-hierarchy.wdio-spec.js:82]: Can you expand more on what we need to wait for sentinel on? There shouldn't be any changes from sentinel processing the docs or the new settings. So we shouldn't need to wait here.

[the remark this answers is not in the document]
Comment (meghna-khemka) [reply tests/e2e/contacts/add-custom-hierarchy.wdio-spec.js:82]: Correct me if I am wrong - when we update settings, Sentinel processes those changes and we need to wait for it complete before proceeding further in the test case to get the updated results. For example using this command: `await sentinelUtils.waitForSentinel();`

[the remark this answers is not in the document]
Comment (newtewt) [reply tests/e2e/contacts/add-custom-hierarchy.wdio-spec.js:82]: You're right that sentinel does process the new changes. However, in this case sentinel won't do anything different from the settings that were applied to the new settings that are being applied.

**Comment by meghna-khemka** [comment]
Looks good to me.Ready to merge.

**Comment by newtewt** [comment]
@meghna-khemka  can you approve?

**Review by meghna-khemka — APPROVED** [review]

**Pull request merged**